ols/app/endpoints/ols.py

```diff
@@ -125,17 +137,27 @@ def retrieve_conversation_id(llm_request: LLMRequest) -> str:
 
 def retrieve_previous_input(user_id: str, llm_request: LLMRequest) -> list[BaseMessage]:
     """Retrieve previous user input, if exists."""
-    previous_input: list[BaseMessage] = []
-    if llm_request.conversation_id:
-        cache_content = config.conversation_cache.get(
-            user_id, llm_request.conversation_id
-        )
-        if cache_content is not None:
-            previous_input = cache_content
-        logger.info(
-            f"{llm_request.conversation_id} Previous conversation input: {previous_input}"
+    try:
+        previous_input: list[BaseMessage] = []
+        if llm_request.conversation_id:
+            cache_content = config.conversation_cache.get(
+                user_id, llm_request.conversation_id
+            )
+            if cache_content is not None:
+                previous_input = cache_content
+            logger.info(
+                f"{llm_request.conversation_id} Previous conversation input: {previous_input}"
+            )
+        return previous_input
+    except Exception as e:
+        logger.error(f"Error retrieving previous user input for user {user_id}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={
+                "response": "Error retrieving previous user input",
```

Comment:
it seems less confusing "Error retrieving conversation history"

Reply:
indeed

Reply:
updated